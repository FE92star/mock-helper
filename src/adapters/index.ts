export function apiAdapters(data: { res_body: any; [key: string]: any }) {
  return {
    ...data,
    response: JSON.parse(data.res_body) || {},
  }
}
