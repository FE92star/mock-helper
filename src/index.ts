import { type ExtensionContext, commands, window } from 'vscode'

export function activate(ctx: ExtensionContext) {
  const disposable = commands.registerCommand('extension.helloWorld', () => {
    window.showInformationMessage('Hello world')
  })

  ctx.subscriptions.push(disposable)
}

export function deactivate() {

}
