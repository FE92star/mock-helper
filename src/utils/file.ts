import fs from 'node:fs'

export function readJsonFile(file: string): Promise<string> {
  return new Promise((resolve, reject) => {
    try {
      const fileContents = fs.readFileSync(file, { encoding: 'utf-8' })

      resolve(fileContents)
    }
    catch (error) {
      reject(error)
    }
  })
}

export function writeJsonFile(file: string, json: any) {
  return new Promise((resolve, reject) => {
    try {
      fs.writeFileSync(file, JSON.stringify(json, null, 2), {
        encoding: 'utf-8',
      })

      resolve(file)
    }
    catch (error) {
      reject(error)
    }
  })
}
