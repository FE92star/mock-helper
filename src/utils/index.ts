import { APP_NAME_PREFIX } from '../adapters'

export function warn(msg: string) {
  return `${APP_NAME_PREFIX}${msg}`
}

export function formatDate(timestamp: number) {
  const date = new Date(timestamp)
  const year = date.getFullYear()
  const monthNum = date.getMonth() + 1
  const month = monthNum > 9 ? monthNum : `0${monthNum}`
  const day = date.getDate() > 9 ? date.getDate() : `0${date.getDate()}`

  return `${year}-${month}-${day}`
}
