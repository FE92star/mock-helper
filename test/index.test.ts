import { describe, expect, it } from 'vitest'
import { type Key, pathToRegexp } from 'path-to-regexp'
import { formatDate } from '../src/utils'

const keys: Key[] = []
const regexp = pathToRegexp('/foo/:bar', keys)

describe('should', () => {
  it('exported', () => {
    expect(1).toEqual(1)
  })
})

describe('date', () => {
  it('format', () => {
    expect(formatDate(1694217600000)).toBe('2023-09-09')
  })
})

describe('query', () => {
  it('string', () => {
    expect(keys).toMatchSnapshot()
    expect(regexp).toMatchSnapshot()
  })
})
