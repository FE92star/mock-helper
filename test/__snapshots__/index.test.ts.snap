// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`query > string 1`] = `
[
  {
    "modifier": "",
    "name": "bar",
    "pattern": "[^\\\\/#\\\\?]+?",
    "prefix": "/",
    "suffix": "",
  },
]
`;

exports[`query > string 2`] = `/\\^\\\\/foo\\(\\?:\\\\/\\(\\[\\^\\\\/#\\\\\\?\\]\\+\\?\\)\\)\\[\\\\/#\\\\\\?\\]\\?\\$/i`;
